imageGenSD.py

```diff
@@ -44,41 +54,49 @@
     #issues.  Obviously can't solve all potential problems.
     with open(cfg_path.absolute()) as json_file:
         params = json.load(json_file)
-    
-    dict_path = pl.Path(params['queue_opts']['rand_dict_path'])
-    
-    #This is just an access check and is done early to allow for an exit if
-    #the file has read/access issues.  The Tag Randomizer class will
-    #separately open a copy when needed with a more read-efficent module.
-    if dict_path.is_file():
-        #This is to guarantee there's no confusion about where the dict is.
-        #If run across computers, this will need to be changed.
-        params['queue_opts']['rand_dict_path'] = dict_path.absolute()
-        #While it would be ideal to simply rely on a user-supplied size, we
-        #can't assume the user will think to do this nor give an accurate
-        #value, hence we have to scan ourselves.
-        params['queue_opts']['dict_size'] = sum(1 for line in open(params['queue_opts']['rand_dict_path']))
-        
-        if int(params['queue_opts']['dict_size']) < int(params['queue_opts']['max_rand_tag_cnt']) or \
-           int(params['queue_opts']['max_rand_tag_cnt']) <= int(params['queue_opts']['min_rand_tag_cnt']):
-            raise IndexError
-    else:
-        raise FileNotFoundError
+
+    dict_path[0] = pl.Path(params['queue_opts']['rand_dict_path'])
+    dict_path[1] = pl.Path(params['profile_opts']['fn_dict_path'])
+    dict_path[2] = pl.Path(params['profile_opts']['ln_dict_path'])
+
+    for dir in range(len(dict_path)):
+        #This is just an access check and is done early to allow for an exit if
+        #the file has read/access issues.  The using classes will separately
+        #open a copy when needed with a more read-efficent module.
+        if not dict_path[dir].is_file():
+            raise FileNotFoundError
+
+    #This is to guarantee there's no confusion about where the dict is.
+    #If run across computers, this will need to be changed.
+    params['queue_opts']['rand_dict_path'] = dict_path[0].absolute()
+    params['profile_opts']['fn_dict_path'] = dict_path[1].absolute()
+    params['profile_opts']['ln_dict_path'] = dict_path[2].absolute()
+
+    #While it would be ideal to simply rely on a user-supplied size, we
+    #can't assume the user will think to do this nor give an accurate
+    #value, hence we have to scan ourselves.
+    params['queue_opts']['dict_size']      = sum(1 for line in open(params['queue_opts']['rand_dict_path']))
```

Comment:
Feels a bit hackish but I guess it works

Reply:
Yeah, I can't necessarily rely on os.stat() and similar due to encoding and cross-platform issues (symlinks), and I don't want to preserve a copy of the dictionary in memory since they're user-configurable and could be massive.  It's always the seemingly simple questions (how many newlines are in this file?) that are hard.